### ISSUE TITLE
Crash during encoding using UTF-16/32 and custom error handler

### ISSUE THREAD

**Issue by 36cd5ce8-7392-4293-b4b5-d518e43b8f56** [python/cpython#81000]
BPO | [36819](https://bugs.python.org/issue36819)
--- | :---
Nosy | @malemburg, @doerwalter, @vstinner, @ezio-melotti, @serhiy-storchaka, @tonybaloney, @ATalaba, @iritkatriel
PRs | <li>python/cpython#13134</li><li>python/cpython#28593</li>
Files | <li>[encode_crash.py](https://bugs.python.org/file48305/encode_crash.py "Uploaded as text/plain at 2019-05-06.20:03:26 by @ATalaba"): Source code to replicate the crash</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/serhiy-storchaka'
closed_at = None
created_at = <Date 2019-05-06.18:51:21.426>
labels = ['interpreter-core', 'type-crash', '3.10', '3.11', 'expert-unicode', '3.9']
title = 'Crash during encoding using UTF-16/32 and custom error handler'
updated_at = <Date 2021-09-29.15:17:06.551>
user = 'https://github.com/ATalaba'
```

bugs.python.org fields:
```python
activity = <Date 2021-09-29.15:17:06.551>
actor = 'doerwalter'
assignee = 'serhiy.storchaka'
closed = False
closed_date = None
closer = None
components = ['Interpreter Core', 'Unicode']
creation = <Date 2019-05-06.18:51:21.426>
creator = 'atalaba'
dependencies = []
files = ['48305']
hgrepos = []
issue_num = 36819
keywords = ['patch']
message_count = 9.0
messages = ['341599', '341658', '402170', '402726', '402826', '402830', '402832', '402833', '402883']
nosy_count = 8.0
nosy_names = ['lemburg', 'doerwalter', 'vstinner', 'ezio.melotti', 'serhiy.storchaka', 'anthonypjshaw', 'atalaba', 'iritkatriel']
pr_nums = ['13134', '28593']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'crash'
url = 'https://bugs.python.org/issue36819'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by alittlesir** [comment]
any question about this issue? if not, i think it is better to close it

**Comment by alittlesir** [comment]
https://github.com/python/cpython/issues/92100, Is this pr cause that the issue cannot be closed?

**Comment by serhiy-storchaka** [comment]
The original issue has been fixed, but we had a discussion about a deeper issue. I opened #96872 for this.

**Comment by vstinner** [comment]
> The original issue has been fixed

Fixed by commit 18b07d773e09a2719e69aeaa925d5abb7ba0c068 GH-28593 according to git bisect :-)
